File: src/main/scala/org/biodatagraphdb/alsdb/value/StringSubjectProperty.scala
package org.biodatagraphdb.alsdb.value

import org.apache.commons.csv.CSVRecord

case class StringSubjectProperty (
               subjectId:String, propertyName:String, propertyValue:String,
               externalSubjectId:String, externalSampleId:String,
               sampleId:Int,sampleType:String, analyteType:String)
{

  def subjectTuple:Tuple2[String,String] = Tuple2(subjectId,externalSubjectId)
}
object StringSubjectProperty extends ValueTrait {
  def parseCSVRecord(record: CSVRecord):StringSubjectProperty  ={
    new StringSubjectProperty(
      record.get("subjectId"),
      record.get("propertyName"), record.get("propertyValue"),
      record.get("externalSubjectId"), record.get("externalSampleId"),
      record.get("sampleId").toInt, record.get("type"),
      record.get("analyteType")
    )
  }
  val columnHeadings:Array[String] = Array("subjectId","propertyName","propertyValue","externalSubjectId",
    "nygcSubjectId","externalSampleId","sampleId"
    ,"type","analyteType")
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/StringSubjectProperty.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class StringSubjectProperty(
        val subjectId: String, val propertyName: String, val propertyValue: String,
        val externalSubjectId: String, val externalSampleId: String,
        val sampleId: Int, val sampleType: String, val analyteType: String
) {
    val subjectTuple = Pair(subjectId,externalSubjectId)
    val columnHeadings = arrayOf("subjectId", "propertyName", "propertyValue", "externalSubjectId",
            "nygcSubjectId", "externalSampleId", "sampleId"
            , "type", "analyteType")

    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): StringSubjectProperty =
                StringSubjectProperty(
                        record.get("subjectId"),
                        record.get("propertyName"),
                        record.get("propertyValue"),
                        record.get("externalSubjectId"),
                        record.get("externalSampleId"),
                        parseValidIntegerFromString(record.get("sampleId")),
                        record.get("type"),
                        record.get("analyteType")
                )
    }
}